mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -117,6 +118,17 @@ case class ScalaPresentationCompiler(
       PcDefinitionProvider(driver, params, search).definitions()
     }
 
+  def documentHighlight(
+      params: OffsetParams
+  ): CompletableFuture[ju.List[DocumentHighlight]] =
+    compilerAccess.withNonInterruptableCompiler(
+      List.empty[DocumentHighlight].asJava,
+      params.token,
+    ) { access =>
+      val driver = access.compiler()
+      PcDocumentHiglightProvider(driver).higlights(params).asJava
```

Comment:
I wonder about this pattern when for every request there is a new class instantiated, wouldn't it be a perfect opportunity to use context functions here? Afterall, those functions depends on some context which is passed by constructor arguments + using this feature is a good example of doogfooding.
```scala
object PcDocumentHiglightProvider {
type WithDriver[T] = InteractiveDriver ?=> T

  def higlights(
      params: OffsetParams
  ): WithDriver[List[DocumentHighlight]] = { driver ?=>
    ???
  }
}
```

Reply:
It could also be an object, which might be less complicated. I don't think we need context functions in this case just yet.